source/attribute/AttributeAccess.h

```diff
@@ -0,0 +1,100 @@
+/* AttributeAccess.h
+Copyright (c) 2023 by tibetiroka
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef ATTRIBUTE_ACCESS_H_
+#define ATTRIBUTE_ACCESS_H_
+
+#include "AttributeCategory.h"
+#include "AttributeEffectType.h"
+
+#include <optional>
+
+class AttributeAccess {
```

Comment:
I'm not sure about the name AttributeAccess. Maybe `AttributeIdentifier`, `AttributeKey` or `AttributeAccessor` would be better? (Since it doesn't control access to attributes, but it is a class to identify an attribute that we want to access.)

Reply:
Maybe we should also reflect in the class-name that this class is only for categorized attributes (`CategorizedAttributeIdentifier` is a bit long, but also very descriptive.)

Or we might want to let all accesses through attributes go through this class (and change AttributeEffectType effect to an union of what is is now and `const char*`). If the category is 0 or nullptr, then this accesses by name, otherwise by category and effect-type.

Reply:
> Or we might want to let all accesses through attributes go through this class (and change AttributeEffectType effect to an union of what is is now and const char*). If the category is 0 or nullptr, then this accesses by name, otherwise by category and effect-type.

I'd rather do that via `AnyAttribute` (which is `std::variant<std::string, AttributeAccess>`, as that's cleaner (and safer, since it's not an union).

> I'm not sure about the name AttributeAccess. Maybe AttributeIdentifier, AttributeKey or AttributeAccessor would be better? (Since it doesn't control access to attributes, but it is a class to identify an attribute that we want to access.)

AttributeAccessor sounds like the best option to me. `CategorizedAttributeIdentifier` is rather long.